test-data/unit/check-enum.test

```diff
@@ -659,6 +659,40 @@ else:
     reveal_type(y)  # No output here: this branch is unreachable
 [builtins fixtures/bool.pyi]
 
+[case testEnumReachabilityChecksWithOrdering]
+from enum import Enum
+from typing_extensions import Literal
+
+class Foo(Enum):
+    _order_ = "A B"
+    A = 1
+    B = 2
+
+Foo._order_  # E: "Type[Foo]" has no attribute "_order_"
```

Comment:
I feel like we could give a better explanation, but the wording is a bit too tricky for me.

Reply:
Yeah this is fine